### PR TITLE
observation documentation

### PR DESCRIPTION
**Pull request by alexholdenmiller**
- adds an entire page dedicated to the observation dict

TBD:
- reference this page from other locations where we talk about this dict
- add more examples of custom fields used in tasks?

### CODE REVIEW

File: docs/source/observations.rst [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  All rights reserved.
+  This source code is licensed under the BSD-style license found in the
+  LICENSE file in the root directory of this source tree. An additional grant
+  of patent rights can be found in the PATENTS file in the same directory.
+
+observations
+============
+
+The primary medium for information flow in ParlAI is a python ``dict``
+containing the actions of an agent (to be observed by a different agent).
```
Comment (jaseweston) [inline docs/source/observations.rst]: by an agent or the environment

File: docs/source/observations.rst [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  Copyright (c) 2017-present, Facebook, Inc.
+  All rights reserved.
+  This source code is licensed under the BSD-style license found in the
+  LICENSE file in the root directory of this source tree. An additional grant
+  of patent rights can be found in the PATENTS file in the same directory.
+
+observations
+============
+
+The primary medium for information flow in ParlAI is a python ``dict``
```
Comment (jaseweston) [inline docs/source/observations.rst]: i like to use the word 'message' as well, maybe work that in somwhere?

File: docs/source/observations.rst [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        'text': 'What is emily afraid of?',
+        'labels': ['mouse'],
+        'label_candidates': ['wolf', 'mouse', 'cat', 'sheep'],
+        episode_done=True,
+    }
+
+
+reward
+------
+This field can be used by reinforcement-learning tasks to send rewards to
```
Comment (jaseweston) [inline docs/source/observations.rst]: 'to other agents' is a bit ambiguous, in the observation dict it is just observed so leave it at that i think?

File: docs/source/observations.rst [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+other agents.
+
+
+image
+-----
+Some datasets may include images. For example, the VQA_v2 dataset contains
```
Comment (jaseweston) [inline docs/source/observations.rst]: Remove the 'some' , or even better just say agents may observe images (doesn't have to be a 'dataset')

**Comment by alexholdenmiller** [comment]
cool yeah and I forgot `eval_labels` (facepalm) last night as well, I'll make these changes

**Pull request merged**